R/bold_italicise_group_labels.R

```diff
@@ -0,0 +1,77 @@
+
+
+bold_italicise_group_labels <-
+  function(x,
+           bold = FALSE,
+           italics = FALSE,
+           print_engine = c("gt", "flextable", "huxtable")) {
+
+    # input checks ---------------------------------------------------------------
+    if (!inherits(x, "gtsummary")) {
+      stop("Class of 'x' must be 'gtsummary'", call. = FALSE)
+    }
+    print_engine <- match.arg(print_engine)
+
+    # apply bold code to table ---------------------------------------------------
+    # THE GROUP COLUMNS WILL ALWAYS BEGIN WITH 'groupname_col*'
+    # APPLY THE FORMATTING TO THOSE COLUMNS
+    if (bold == TRUE) {
+      switch(
+        print_engine,
+        "gt" = {
+          x <- gtsummary::as_gt(x)
+          x <-
+            x %>%
+            gt::tab_options(row_group.font.weight = "bold")
+        },
+        "flextable" = {
+          x <- gtsummary::as_flex_table(x)
+          x <-
+            x %>%
+            flextable::bold(j="groupname_col")
+        },
+        "huxtable" =  {
+          x = gtsummary::as_hux_table(x)
+          x <-
+            x %>%
+            huxtable::set_bold(row = 3:nrow(.)-1, col = 1)
```

Comment:
Can you double check this works when you have a spanning header row? The row numbers change depending on the number of header rows.

Instead of using `col = 1`, can we have it set to identify columns with names that begin with `"groupname_col"`?

Reply:
Regarding spanning headers: you're right, row = 3:nrow(.)-1 doesn't work correctly if table includes a spanning header. i coded a solution, a bit crappy, tho...
The thing is that huxtables are a dataframe with named rows, those that we want to get in italics are numbered, the rest of rows are number with a preceding dot (for example ".1") or some kind of blank (""). 

I did the following:
rowsToItalic <- stringr::str_detect(pattern = "\\.", string = rownames(x), negate = TRUE)

rowsToItalic[1] <- FALSE

But had to add the last line because I couldn't make a regex that detects such a thing like "" (it is not a white space), so the crappy part is to forcedly make FALSE the first element of that vector.

Then rowsToItalic is passed to the huxtable italic function:

huxtable::set_italic(row = rowsToItalic,
                               col = dplyr::starts_with("groupname_col"))

...surely you have a more elegant solution